### PR TITLE
NIFI-10870 Wrapped debug logger statements with if statements in the …

### PR DESCRIPTION
**Pull request by MikeThomsen**
…elasticsearch bundle and added a debug logger for dumping raw elastic queries.

# Summary

[NIFI-00000](https://issues.apache.org/jira/browse/NIFI-00000)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/QueryElasticsearchHttp.java
Comment (ChrisSamo632) [inline nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/QueryElasticsearchHttp.java:467]: Should use a log formatting String instead, like other logger calls (rther than String concatenation)

File: nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractPutElasticsearch.java
Comment (ChrisSamo632) [inline nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractPutElasticsearch.java:169]:
```suggestion
            } else if (getLogger().isDebugEnabled()) {
                    getLogger().debug(output);
            }
```

File: nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java
Comment (ChrisSamo632) [inline nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java:809]: Suggest creating the `Writer` a single time, e.g. in `onEnabled` after the `mapper` has been initialised

Also, `builder` and `out` can be made `final`

**Review by ChrisSamo632 — CHANGES_REQUESTED** [review]
Thanks for the contribution @MikeThomsen , just a few minor suggestions from me

Personally I'm unconvinced for the nee of the `isLevelEnabled` check around any non-complex logging statements as the first thing that the Logger does in any `debug` method call is check whether `DEBUG` is enabled (i.e. effectively calling `isDebugEnabled` internally), so actually you're duplicating effort in a lot of these cases, albeit a very small overhead and I'll not reject the changes. There are a couple of instances (particularly the Request logging that's been added) that do benefit from the `isDebugEnabled` check certainly and your changes have fixed some logging statements that were still using the old `Object[]` wrapper approach or unnecessary String concatenation

File: nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java
Comment (ChrisSamo632) [inline nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java:800]:
```suggestion
            StringBuilder builder = new StringBuilder(1000);
```

Initialising the `StringBuilder` with a large(r) buffer is sensible when we expect to be putting a lot of characters in the String - it's more performant (marginal gains, apprectiating this is only being called if DEBUG logging is enabled for the class)

File: nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java
Comment (MikeThomsen) [reply nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientServiceImpl.java:809]: Making the change, but noting here that we both seem to agree that according to the docs it should be thread-safe:

> Builder object that can be used for per-serialization configuration of serialization parameters, such as JSON View and root type to use. (and thus fully thread-safe with no external synchronization); new instances are constructed for different configurations. Instances are initially constructed by [ObjectMapper](https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/ObjectMapper.html) and can be reused in completely thread-safe manner with no explicit synchronization

[Source](https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/ObjectWriter.html)

**Comment by MikeThomsen** [comment]
@ChrisSamo632 Updated.

**Review by ChrisSamo632 — APPROVED** [review]
Thanks for the contribution @MikeThomsen, the changes LGTM 👍, will merge pending successful build

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
@MikeThomsen and @ChrisSamo632, the addition if debug logging in `ElasticSearchClientServiceImpl` looks good, but the addition of conditional `isDebugEnabled()` checks elsewhere seem unnecessary. Many of the log statements do not have any parameters, and those that have parameters do not perform additional computation.

I would like to see most of the additional `isDebugEnabled()` checks removed before merging these changes. A couple might be reasonable, but most seem unnecessary.

File: nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractPutElasticsearch.java
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractPutElasticsearch.java:168]: Aside from the conditional `isDebugEnabled()` in the service implementation that logs the HTTP entity, this is the only other conditional that seems necessary.

Comment (ChrisSamo632) [reply nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractPutElasticsearch.java:168]: Ah, yeah, forgot to follow up on that form my original comment (https://github.com/apache/nifi/pull/6712#pullrequestreview-1192230945), good catch

**Comment by MikeThomsen** [comment]
Going to close this and start over.

**Pull request closed without merging**
